java/src/main/java/com/powsybl/python/network/NetworkCFunctions.java

```diff
@@ -166,6 +163,46 @@ public static ObjectHandle loadNetworkFromString(IsolateThread thread, CCharPoin
         });
     }
 
+    @CEntryPoint(name = "loadNetworkFromBinaryBuffers")
+    public static ObjectHandle loadNetworkFromBinaryBuffers(IsolateThread thread, CCharPointerPointer data, CIntPointer dataSizes, int bufferCount, CCharPointerPointer parameterNamesPtrPtr,
+                                                           int parameterNamesCount, CCharPointerPointer parameterValuesPtrPtr, int parameterValuesCount, ObjectHandle reporterHandle,
+                                                           ExceptionHandlerPointer exceptionHandlerPtr) {
+        return doCatch(exceptionHandlerPtr, () -> {
+            Properties parameters = createParameters(parameterNamesPtrPtr, parameterNamesCount, parameterValuesPtrPtr, parameterValuesCount);
+            Reporter reporter = ObjectHandles.getGlobal().get(reporterHandle);
+            List<Integer> bufferSizes = CTypeUtil.toIntegerList(dataSizes, bufferCount);
+            List<ReadOnlyDataSource> dataSourceList = new ArrayList<>();
+            for (int i = 0; i < bufferCount; ++i) {
+                ByteBuffer buffer = CTypeConversion.asByteBuffer(data.read(i), bufferSizes.get(i));
+                Optional<CompressionFormat> format = detectCompressionFormat(buffer);
+                if (format.isPresent() && CompressionFormat.ZIP.equals(format.get())) {
```

Comment:
We could also support the no compression case ?

Reply:
I don't think so, you are going to run again on the initial problem of not having an extension to try to guess the format. With a zip format you can try to access the files inside with a basename but it is still the extension that is used to select an importer.

Reply:
yes you are right